main.go

```diff
@@ -0,0 +1,520 @@
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//db.deviceReport.createIndex({name: 1}, {unique: true});
+
+	//db.createCollection("deviceService");
+	deviceService := mgo.Collection{
+		Database: &db,
+		Name:     "deviceService",
+		FullName: "db.deviceService",
+	}
+	err = deviceService.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//db.deviceService.createIndex({name: 1}, {unique: true});
+
+	//db.createCollection("provisionWatcher");
+	provisionWatcher := mgo.Collection{
+		Database: &db,
+		Name:     "provisionWatcher",
+		FullName: "db.provisionWatcher",
+	}
+	err = provisionWatcher.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//db.provisionWatcher.createIndex({name: 1}, {unique: true});
+
+	//db.createCollection("schedule");
+	schedule := mgo.Collection{
+		Database: &db,
+		Name:     "schedule",
+		FullName: "db.schedule",
+	}
+	err = schedule.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//db.schedule.createIndex({name: 1}, {unique: true});
+
+	//db.createCollection("scheduleEvent");
+	scheduleEvent := mgo.Collection{
+		Database: &db,
+		Name:     "scheduleEvent",
+		FullName: "db.scheduleEvent",
+	}
+	err = scheduleEvent.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//db.scheduleEvent.createIndex({name: 1}, {unique: true});
+	//
+
+	//db=db.getSiblingDB('coredata')
+	db = mgo.Database{
+		Session: session,
+		Name:    "coredata",
+	}
+	//db.createUser({ user: "core",
+	//pwd: "password",
+	//roles: [
+	//{ role: "readWrite", db: "coredata" }
+	//]
+	//});
+	err = db.UpsertUser(&mgo.User{
+		Username: "core",
+		Password: "password",
+		Roles: []mgo.Role{
+			mgo.RoleReadWrite,
+		},
+	})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//db.createCollection("event");
+	event := mgo.Collection{
+		Database: &db,
+		Name:     "event",
+		FullName: "db.event",
+	}
+	err = event.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//TODO what
+	//db.event.createIndex({"device": 1}, {unique: false});
+
+	//db.createCollection("reading");
+	reading := mgo.Collection{
+		Database: &db,
+		Name:     "reading",
+		FullName: "db.reading",
+	}
+	err = reading.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+
+	//db.createCollection("valueDescriptor");
+	valueDescriptor := mgo.Collection{
+		Database: &db,
+		Name:     "valueDescriptor",
+		FullName: "db.valueDescriptor",
+	}
+	err = valueDescriptor.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		println("Error during execution: " + err.Error())
+	}
+	//TODO what
+	//db.reading.createIndex({"device": 1}, {unique: false});
+	//db.valueDescriptor.createIndex({name: 1}, {unique: true});
```

Comment:
Creation of indexes is commented out. I can't find implementation to handle these.